rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -633,28 +561,20 @@ void FleetUpdateHandle::Implementation::dispatch_request_cb(
       return;
     }
 
-    auto request_to_cancel_it = assigned_requests.find(id);
-    if (request_to_cancel_it == assigned_requests.end())
+    if (!is_queued(id))
```

Comment:
The proposed change alters the intention of the code. The original approach checks for whether or not the task to be cancelled was assigned to this fleet (whether queued or active). However the new approach only checks for queued tasks. If the task is currently active, it will exit after printing ```Unable to cancel task with task_id:[%s] as it is not queued to fleet:[]```. But this is not the case. We would want the execution to reach the block ```RCLCPP_WARN(
        node->get_logger(),
        "Unable to cancel active task with task_id:[%s]. Only queued tasks may "
        "be cancelled.",
        id.c_str());``` below